### ISSUE TITLE
Working on an identical project `minbpe.rs`

### ISSUE THREAD

**Issue by shubham0204** [gnp/minbpe-rs#1]
I was working on building the Rust implementation for `karpathy/minbpe` and realized that there was a crate with the same name while uploading my crate. 

Here's my implementation: [shubham0204/minbpe.rs](https://github.com/shubham0204/minbpe.rs)

It has an implementation for `BasicTokenizer` for now, and `RegexTokenizer` and `GPT4Tokenizer` have to be implemented. I can contribute the following to your implementation:

1. `README.md` with usage instructions and sample code
2. Unit tests for functions in `basic.rs` and `base.rs`

I've added these in my implementation, so I can easily contribute these to your project.

**Comment by gnp** [comment]
Hi! Pleased to meet you. You are right -- I didn't invest in a better README.md, yet. I did add doc comment examples to a few things though, which could be source material for README.md contents.

In any case I'd be pleased to accept PRs for adding such things (and of course fixes if I got anything wrong!).

My objective was to make something pretty much 1-1 with what Andrej did as much as possible, but using Rust idioms.

And, I wanted to be able to have each of the implementations of tokenizers be in their own feature, so that if someone wanted to use use the base trait for Tokenizer, Loadable, Saveable, Trainable, Token, etc. they could do that and experiment making their own tokenizers but leveraging that basis.
